client/client.go

```diff
@@ -361,6 +350,17 @@ func Configure(logger hclog.Logger, providerConfig interface{}) (schema.ClientMe
 			return nil, fmt.Errorf(awsFailedToConfigureErrMsg, accountID, err, checkEnvVariables())
 		}
 
+		if account.RoleARN != "" {
```

Comment:
maybe put this in the `switch` or new `if` `else` ?

Reply:
Not sure this comment is still relevant after the refactoring...